metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/MathFunctions.java

```diff
@@ -220,4 +219,156 @@ public Object apply(List<Object> args) {
       }
     }
   }
+
+  @Stellar(namespace = "BOYERMOORE"
+      , name = "ADD"
+      , description = "Adds value to a Boyer-Moore list. [Boyer-Moore](https://en.wikipedia.org/wiki/Boyer%E2%80%93Moore_majority_vote_algorithm)"
+      , params = {
+        "state - state holder for list of values. If null, add will initialize a new state value.",
+        "value(s) - single object or list of values to add to the state object."
+      }
+      , returns = "Current state of the Boyer-Moore algorithm representing the current value that"
+      + "holds a plurality across all values added thus far."
+  )
+  public static class BoyerMooreAdd extends BaseStellarFunction {
+
+    @Override
+    public Object apply(List<Object> args) {
+      if (args.size() < 2) {
+        throw new IllegalArgumentException(
+            "Must pass an initial state (may be null) and at least one value to add to the list");
+      } else {
+        BoyerMooreState state = ConversionUtils.convert(args.get(0), BoyerMooreState.class);
+        if (state == null) {
+          state = new BoyerMooreState();
+        }
+        Object secondArg = args.get(1);
+        if (secondArg instanceof List) {
+          state.addAll((List) secondArg);
+        } else {
+          state.add(secondArg);
+        }
+        return state;
+      }
+    }
+  }
+
+  public static class BoyerMooreState {
+    private Long counter;
+    private Object m;
+
+    public BoyerMooreState() {
+      counter = 0L;
+    }
+
+    public BoyerMooreState(Optional<List<BoyerMooreState>> previousStates, Optional<BoyerMooreState> currentState) {
+      this();
+      currentState.ifPresent(boyerMooreState -> {
+        m = boyerMooreState.getPlurality();
+        counter = boyerMooreState.getCounter();
+      });
+      for (BoyerMooreState state : previousStates.orElse(new ArrayList<>())) {
+        Object plurality = state.getPlurality();
+        Long pluralityCount = state.getCounter();
+        add(plurality, pluralityCount);
+      }
+    }
+
+    public Object add(Object item) {
+      if (item != null) {
+        if (counter == 0) {
+          m = item;
+          counter = 1L;
+        } else if (item.equals(m)) {
+          counter++;
+        } else {
+          counter--;
+        }
+      }
+      return m;
+    }
+
+    public Object add(Object item, Long counter) {
+      if (item != null) {
+        if (this.counter == 0) {
+          m = item;
+          this.counter = counter;
```

Comment:
So, Stellar will parse and create Numbers, I'm wondering if there is a possibility of strange errors or rounding here.  Like there are two not equal numbers but they end up equal through the conversion process.  Does that make sense?

Reply:
It's true, I make no effort to massage data types here - if you want to interpret a 1 as a string "1", then they will be considered distinct. I didn't have a suitable mechanism for determining types. I thought it was reasonable for a user to be able to make that type cast as part of their Stellar expression building, but I'm certainly open to suggestions.

Reply:
AFA rounding errors - the counter you're seeing there are internal state. In the case of merging multiple states together, it will effectively merge the counts to nominate a new plurality leader. If we could do this in 2 passes, you'd have a guaranteed majority, but I thought it was still useful to see what item has the most without necessarily being a majority. I'll add some use case documentation this week as I think this might be useful in the profiler.

Reply:
no, i'm talking about  object.equals(object2)  when the type of the objects is determined by antlr etc

Reply:
I know we do this in a number of places where the Objects provided via the inbound List are handed off directly to a java.util.List without any casting or special parsing, e.g. 
* BloomFilter -> https://github.com/apache/metron/blob/master/metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/DataStructureFunctions.java#L48
* Data structure functions such as List -> https://github.com/apache/metron/blob/master/metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/dsl/functions/DataStructureFunctions.java#L189
* HLLP -> https://github.com/apache/metron/blob/master/metron-analytics/metron-statistics/src/main/java/org/apache/metron/statistics/approximation/HyperLogLogPlusFunctions.java#L49

Is that what you're asking?

Reply:
No, more of something that is prob. an edge case, such as fields from input that are not normalized, so they come in as 1 or 1.0 or something and will be unequal.